### PR TITLE
Count number of banned messages implemented

### PR DESCRIPTION
**Pull request by marclll**

### CODE REVIEW

File: src/main/java/org/pesmypetcare/webservice/dao/communitymanager/ForumDaoImpl.java [hunk 1]
```diff
@@ -187,24 +187,47 @@ public void reportMessage(String parentGroup, String forumName, String creator,
...
+            batch.update(messageSnapshot.getReference(), REPORTED_BY_FIELD, messages);
+        } else {
+            if (messages.contains(reporter)) {
+                throw new InvalidOperationException("409", "This user already reported the message");
+            }
+            batch.update(messageSnapshot.getReference(), REPORTED_BY_FIELD, FieldValue.arrayUnion(reporter));
+            if (messages.size() == COUNTER) {
+                batch.update(messageSnapshot.getReference(), BANNED_FIELD, true);
+                String usernamePath = Path.ofDocument(Collections.used_usernames, creator);
+                String uid = documentAdapter.getStringFromDocument(usernamePath, "user");
```
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/dao/communitymanager/ForumDaoImpl.java:203]: Codacy Issue found: [The String "user" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5638522)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 6
           

Coverage decreased per file
===========================
- src/main/java/org/pesmypetcare/webservice/dao/communitymanager/ForumDaoImpl.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5638522&bid=18465494)

File: src/test/java/org/pesmypetcare/webservice/dao/communitymanager/ForumDaoTest.java [hunk 2]
```diff
@@ -421,17 +421,18 @@ public void reportMessage() throws DatabaseAccessException, DocumentException,
...
                     verify(batch).update(same(documentReference), eq("reportedBy"),
                         eq(FieldValue.arrayUnion(username2)));
-                    verify(batch).update(same(documentReference), eq("banned"),
-                        eq(true));
+                    verify(batch).update(same(documentReference), eq("banned"), eq(true));
                 }
 
                 @Test
-                public void unbanMessage() throws DatabaseAccessException, DocumentException {
+                public void unbanMessage() throws DatabaseAccessException, DocumentException, InvalidOperationException {
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/dao/communitymanager/ForumDaoTest.java:428]: Codacy Issue found: [Line is longer than 120 characters (found 121).](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5638522)

File: src/main/java/org/pesmypetcare/webservice/service/communitymanager/ForumServiceImpl.java [hunk 3]
```diff
@@ -7,6 +7,7 @@
 import org.pesmypetcare.webservice.error.DatabaseAccessException;
 import org.pesmypetcare.webservice.error.DocumentException;
 import org.pesmypetcare.webservice.error.InvalidOperationException;
+import org.pesmypetcare.webservice.utilities.UTCLocalConverter;
```
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/service/communitymanager/ForumServiceImpl.java:10]: Codacy Issue found: [Unused import - org.pesmypetcare.webservice.utilities.UTCLocalConverter.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5638522)

File: src/main/java/org/pesmypetcare/webservice/dao/communitymanager/ForumDaoImpl.java [hunk 4]
```diff
@@ -187,24 +187,47 @@ public void reportMessage(String parentGroup, String forumName, String creator,
...
+                String path = Path.ofDocument(Collections.users, uid);
+                Long BanCounter = (Long) documentAdapter.getDocumentField(path, "messagesBanned");
```
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/dao/communitymanager/ForumDaoImpl.java:205]: Codacy Issue found: [Name 'BanCounter' must match pattern '^[a-z][a-zA-Z0-9]*$'.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5638522)

Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/dao/communitymanager/ForumDaoImpl.java:205]: Codacy Issue found: [The String "messagesBanned" appears 4 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5638522)

File: src/test/java/org/pesmypetcare/webservice/dao/communitymanager/ForumDaoTest.java [hunk 5]
```diff
@@ -421,17 +421,18 @@ public void reportMessage() throws DatabaseAccessException, DocumentException,
                             eq("creator"), eq(username), eq(publicationDate), eq(date));
                     verify(batch).update(same(documentReference), eq("reportedBy"),
                         eq(FieldValue.arrayUnion(username2)));
-                    verify(batch).update(same(documentReference), eq("banned"),
-                        eq(true));
+                    verify(batch).update(same(documentReference), eq("banned"), eq(true));
```
Comment (santidrj) [inline src/test/java/org/pesmypetcare/webservice/dao/communitymanager/ForumDaoTest.java:424]: Codacy Issue found: [The String "banned" appears 2 times in the file.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5638522)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5638522&bid=18465494)

File: src/main/java/org/pesmypetcare/webservice/dao/communitymanager/ForumDaoImpl.java [hunk 6]
```diff
@@ -187,24 +188,48 @@ public void reportMessage(String parentGroup, String forumName, String creator,
...
+        if (isMessageBanned == null || !isMessageBanned) {
+            throw new InvalidOperationException("409", "The message wasn't banned");
+        }
         WriteBatch batch = documentAdapter.batch();
         batch.update(messageSnapshot.getReference(), REPORTED_BY_FIELD, new ArrayList<String>());
         batch.update(messageSnapshot.getReference(), BANNED_FIELD, false);
+        String usernamePath = Path.ofDocument(Collections.used_usernames, creator);
+        String uid = documentAdapter.getStringFromDocument(usernamePath, "user");
+        String path = Path.ofDocument(Collections.users, uid);
+        Long BanCounter = (Long) documentAdapter.getDocumentField(path, MESSAGES_BANNED_FIELD);
```
Comment (santidrj) [inline src/main/java/org/pesmypetcare/webservice/dao/communitymanager/ForumDaoImpl.java:231]: Codacy Issue found: [Name 'BanCounter' must match pattern '^[a-z][a-zA-Z0-9]*$'.](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5638522)

**Comment by santidrj** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Coverage decreased per file
===========================
- src/main/java/org/pesmypetcare/webservice/dao/communitymanager/ForumDaoImpl.java  -1
         
```


See the complete overview on [Codacy](https://app.codacy.com/gh/Grupo13-PES-Mascotas/PES_My-Pet-Care-Webservice/pullRequest?prid=5638522&bid=18465494)

**Review by oricat8 — APPROVED** [review]

**Review by santidrj — APPROVED** [review]
:+1:

**Pull request merged**